druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/JDBCExtractionNamespaceCacheFactory.java

```diff
@@ -96,73 +89,146 @@ Void populateRowListFromJDBC(
+        return populateRowListFromJDBC(extractionNamespace, query, lastDBUpdate, handle, rm, "");
+    }
+
+    Void populateRowListFromJDBC(
+            JDBCExtractionNamespace extractionNamespace,
+            String query,
+            Timestamp lastDBUpdate,
+            Handle handle,
+            RowMapper rm,
+            String whereClauseExtension
     ) {
         Timestamp updateTS;
         if (extractionNamespace.isFirstTimeCaching()) {
             extractionNamespace.setFirstTimeCaching(false);
-            query = String.format("%s %s", query, FIRST_TIME_CACHING_WHERE_CLAUSE);
+            query = String.format("%s %s %s",
+                    query,
+                    getBaseWhereClause(FIRST_TIME_CACHING_WHERE_CLAUSE, extractionNamespace),
+                    whereClauseExtension
+            );
             updateTS = lastDBUpdate;
 
         } else {
-            query = String.format("%s %s", query, SUBSEQUENT_CACHING_WHERE_CLAUSE);
+            query = String.format("%s %s %s",
+                    query,
+                    getBaseWhereClause(SUBSEQUENT_CACHING_WHERE_CLAUSE, extractionNamespace),
+                    whereClauseExtension
+            );
             updateTS = extractionNamespace.getPreviousLastUpdateTimestamp();
         }
 
         List<Void> p = handle.createQuery(query).map(rm)
                 .setFetchSize(FETCH_SIZE)
-                .bind("lastUpdatedTimeStamp", updateTS)
+                .bind("lastUpdatedTimeStamp",
+                        getTsValue(extractionNamespace, updateTS))
                 .list();
 
         return null;
     }
 
+    protected String getWhereClauseExtension(String id, JDBCExtractionNamespace extractionNamespace) {
+        String whereClauseExtension = "";
+        if (extractionNamespace.hasTsColumnConfig() && extractionNamespace.getTsColumnConfig().hasSecondaryTsColumn()) {
+            String maxVal = (String) getMaxValFromColumn(id, extractionNamespace, StringMapper.FIRST, extractionNamespace.getTsColumnConfig().getSecondaryTsColumn(), extractionNamespace.getTable());
+            whereClauseExtension = String.format(WHERE_CLAUSE_EXTENSION,
+                    extractionNamespace.getTsColumnConfig().getSecondaryTsColumn(),
+                    extractionNamespace.getTsColumnConfig().getSecondaryTsColumnCondition(),
+                    StringUtil.quoteStringLiteral(maxVal)
+            );
+        }
+        return whereClauseExtension;
+    }
+
+    private Object getTsValue(JDBCExtractionNamespace extractionNamespace, Timestamp updateTS) {
+        Object tsValue = updateTS;
+        if (extractionNamespace.hasTsColumnConfig()) {
+            if (extractionNamespace.getTsColumnConfig().isVarchar()) {
+                tsValue = new SimpleDateFormat(extractionNamespace.getTsColumnConfig().getFormat()).format(updateTS);
+            } else if (extractionNamespace.getTsColumnConfig().isBigint()) {
+                tsValue = updateTS.getTime();
+            }
```

Comment:
Let's add an else here with a warning of unknown type

Reply:
sure, will do

Reply:
throwing an exception instead if it's an unknown type